metadata/errors.go

```diff
@@ -1,22 +1,81 @@
 package metadata
 
-import "github.com/pkg/errors"
+type metadataErr struct {
+	exists   bool
+	notFound bool
+	notEmpty bool
+	desc     string
+}
+
+// ErrExists is returned when an item already exists in metadata
+func ErrExists(msg string) error {
+	if msg == "" {
+		msg = "metadata: exists"
+	}
+	return metadataErr{
+		exists: true,
+		desc:   msg,
+	}
+}
+
+// ErrNotFound is returned when an item cannot be found in metadata
+func ErrNotFound(msg string) error {
+	if msg == "" {
+		msg = "metadata: not found"
+	}
+	return metadataErr{
+		notFound: true,
+		desc:     msg,
+	}
+}
+
+// ErrNotEmpty is returned when a metadata item can't be deleted because it is not empty
+func ErrNotEmpty(msg string) error {
+	if msg == "" {
+		msg = "metadata: namespace not empty"
+	}
+	return metadataErr{
+		notEmpty: true,
+		desc:     msg,
+	}
+}
 
-var (
-	ErrExists   = errors.New("metadata: exists")
-	ErrNotFound = errors.New("metadata: not found")
-	ErrNotEmpty = errors.New("metadata: namespace not empty")
-)
+func (m metadataErr) Error() string {
+	return m.desc
+}
+
+func (m metadataErr) Exists() bool {
+	return m.exists
+}
+
+func (m metadataErr) NotFound() bool {
```

Comment:
Errors should not implement the `NotFound()` method unless they are a not found error. For example, we want to be able to do this:

```
switch err := errors.Cause(err).(type) {
case interface{ NotFound() bool }:
default:
 // another error type.
}
```